lib/plugins/wallet/coinbasepro/coinbasepro.js

```diff
@@ -0,0 +1,163 @@
+  return checkCryptoCode(cryptoCode)
+    .then(() => getWallet(account, cryptoCode))
+    .then(_.get('balance'))
+}
+
+function newAddress (account, info, tx, settings, operatorId) {
+  return checkCryptoCode(info.cryptoCode)
+    .then(() => getWallet(account, info.cryptoCode))
+    .then(wallet => {
+      return request({
+        path: `/coinbase-accounts/${wallet.id}/addresses`,
+        method: 'POST'
+      }, account)
+        .then(_.get('address'))
+    })
+}
+
+function convertPendingBalance (ratesRec, balance, cryptoCode) { 
+      const rates = ratesRec.rates
+      const rate = (rates.ask.plus(rates.bid)).div(2)
+      const cryptoRec = coinUtils.getCryptoCurrency(cryptoCode)
+      const unitScale = cryptoRec.unitScale
+      const baseUnits = new BN(1).shiftedBy(unitScale)
+      const result = baseUnits.times(balance).div(rate)
+      return result
+}
+
+function getRates(fiatCode, cryptoCode) {
+  return settingsLoader.loadLatest().then(settings => ticker.getRates(settings, fiatCode, cryptoCode))
+}
+
+function newFunding (account, cryptoCode, fiatCode, settings, operatorId) {
+  return checkCryptoCode(cryptoCode)
+    .then(() => {
+      const promises = [
+        getWallet(account, cryptoCode),
+        newAddress(account, { cryptoCode }),
+        getRates(fiatCode, cryptoCode)
+      ]
+      return Promise.all(promises)
+    })
+    .then(([wallet, fundingAddress, ratesRec]) => ({
+      fundingConfirmedBalance: new BN(wallet.balance),
+      fundingPendingBalance: convertPendingBalance(ratesRec, new BN(wallet['hold_balance']), cryptoCode),
+      fundingAddress
+    }))
+}
+
+function sendCoins (account, tx, settings, operatorId) {
+  const { toAddress, cryptoAtoms, cryptoCode } = tx
+  return checkCryptoCode(cryptoCode)
+    .then(() => {
+      const body = {
+        crypto_address: toAddress,
+        amount: cryptoAtoms.toNumber(),
+        currency: cryptoCode
+      }
+      return request({
+        path: '/withdrawals/crypto',
+        method: 'POST',
+        body
+      }, account)
+    })
+    .then(result => {
+      let fee = parseFloat(result.fee)
+      let txid = result.id
+
+      return { txid: txid, fee: new BN(fee).decimalPlaces(0) }
+    })
+    .catch(err => {
+      if (err.message === 'insufficient funds') throw new E.InsufficientFundsError()
+      throw err
+    })
+}
+
+function getStatus () {
+  return Promise.reject('Cash-out not supported!')
+}
+
+function getWallet (account, cryptoCode) {
+  return request({ path: '/coinbase-accounts', method: 'GET' }, account)
+    .then(res => _.find(it => it.currency === cryptoCode)(res))
+}
+
+function checkCryptoCode (cryptoCode) {
+  if (!SUPPORTED_COINS.includes(cryptoCode)) {
+    return Promise.reject(new Error('Unsupported crypto: ' + cryptoCode))
+  }
+
+  return Promise.resolve()
+}
+
+function checkBlockchainStatus (cryptoCode) {
+  return checkCryptoCode(cryptoCode)
+    .then(() => Promise.resolve('ready'))
+}
```

Comment:
The `Promise.resolve()` is already implicit

```suggestion
function checkBlockchainStatus (cryptoCode) {
  return checkCryptoCode(cryptoCode)
    .then(() => 'ready')
}
```